### ISSUE TITLE
Zone parses panic when parsing a zone that contains a hostname with a leading whitespace

### ISSUE THREAD

**Issue by wuerges** [bluejekyll/trust-dns#1843]
**Describe the bug**
Zone parses panic when parsing a zone that contains a hostname with a leading whitespace, instead of resulting in an `Err`.

**To Reproduce**
This PR contains a test that shows the bug:
https://github.com/bluejekyll/trust-dns/pull/1842

**Expected behavior**
When parsing an invalid zone, we expected an `Err`, but instead trust-dns panics:

```
---- serialize::txt::zone::tests::test_zone_parse stdout ----
thread 'serialize::txt::zone::tests::test_zone_parse' panicked at 'assertion failed: str.chars().all(|x| char::is_ascii_uppercase(&x))', crates/proto/src/rr/dns_class.rs:56:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**System:**
 - OS: macOs
 - Architecture: intel 
 - Version: main
 - rustc version: 1.65.0

**Version:**
Crate: client
Version: main

**Additional context**

**Comment by wuerges** [comment]
As @darnuria pointed out, the issue only happens in the `test` configuration. 
However, even so, the PR makes a small tweak to the `debug_assert` that caused the panic so this panic is much more unlikely.

**Comment by bluejekyll** [comment]
I believe this is resolved now in #1842
